### PR TITLE
Support (wildcard) param groups in initialization

### PR DESCRIPTION
**Pull request by ottonemo**

### CODE REVIEW

File: docs/user/neuralnet.rst [hunk 1]
```diff
@@ -114,6 +114,16 @@ optimizer
 This should be a PyTorch optimizer,
 e.g. ``torch.optim.SGD``. After initializing the ``NeuralNet``, the
 initialized optimizer will stored in the ``optimizer_`` attribute.
+During initialization you can already define param groups to define
+different parametrizations for certain parameters identified by
+strings with globbing:
```
Comment (benjamin-work) [inline docs/user/neuralnet.rst]: This sentence is not very clear, maybe you could extend it a little bit (e.g. by giving an example or providing a link to the pytorch docs).

Comment (ottonemo) [reply docs/user/neuralnet.rst]: I wasn't sure if this is the best place to document this in detail. I can improve upon the existing text by adding some more details but I have the feeling that this is not the best place to explain param groups in all their glory.

Comment (benjamin-work) [reply docs/user/neuralnet.rst]: Nonetheless, I find the sentence a little awkward. Maybe read it out loud and you may see what I mean.

**Comment by ottonemo** [comment]
@benjamin-work better?

**Pull request merged**